### PR TITLE
Update core dependency

### PR DESCRIPTION
**Pull request by nilebox**

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
Please run `make gotidy`.

**Comment by bogdandrutu** [comment]
@nilebox ExampleComponents were moved to `componenttest` from `config` package

File: exporter/splunkhecexporter/metricdata_to_splunk.go [hunk 1]
```diff
@@ -213,7 +213,7 @@ func mapSummaryValue(metric *metricspb.Metric, summaryValue *metricspb.SummaryVa
 	if percentiles == nil {
 		return values, fmt.Errorf(
 			"unknown percentiles values for summary metric %q",
-			*metric)
+			metricName)
```
Comment (nilebox) [inline exporter/splunkhecexporter/metricdata_to_splunk.go:216]: @atoulme as the original author of this exporter, please confirm if this fix is correct.

Comment (bogdandrutu) [reply exporter/splunkhecexporter/metricdata_to_splunk.go:216]: Looks good, it is just an error.

**Comment by bogdandrutu** [comment]
@nilebox there are few more fixes to pass tests :)

**Comment by bogdandrutu** [comment]
testutils/metrics.go:25:50: copylocks: AssertResource passes lock by value: github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1.Resource contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
func AssertResource(t *testing.T, actualResource resourcepb.Resource,
                                                 ^
testutils/metrics.go:40:57: copylocks: AssertMetricsWithLabels passes lock by value: github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1.Metric contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
func AssertMetricsWithLabels(t *testing.T, actualMetric metricspb.Metric,
                                                        ^
testutils/metrics.go:52:16: copylocks: call of getLabelsMap copies lock value: github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1.Metric contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
		getLabelsMap(actualMetric),
		             ^
testutils/metrics.go:56:19: copylocks: call of AssertMetrics copies lock value: github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1.Metric contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
	AssertMetrics(t, actualMetric, expectedMetric, expectedType, expectedValue)
	                 ^
testutils/metrics.go:59:47: copylocks: AssertMetrics passes lock by value: github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1.Metric contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
func AssertMetrics(t *testing.T, actualMetric metricspb.Metric,
                                              ^
testutils/metrics.go:88:21: copylocks: getLabelsMap passes lock by value: github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1.Metric contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
func getLabelsMap(m metricspb.Metric) map[string]string {
                    ^
collection/cronjobs_test.go:38:30: copylocks: call of testutils.AssertResource copies lock value: github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1.Resource contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
	testutils.AssertResource(t, *actualResourceMetrics[0].resource, k8sType,
	                            ^
collection/cronjobs_test.go:47:29: copylocks: call of testutils.AssertMetrics copies lock value: github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1.Metric contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
	testutils.AssertMetrics(t, *actualResourceMetrics[0].metrics[0], "k8s/cronjob/active_jobs",
	                           ^
collection/daemonsets_test.go:38:30: copylocks: call of testutils.AssertResource copies lock value: github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1.Resource contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
	testutils.AssertResource(t, *rm.resource, k8sType,
	                            ^
collection/daemonsets_test.go:47:29: copylocks: call of testutils.AssertMetrics copies lock value: github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1.Metric contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
	testutils.AssertMetrics(t, *rm.metrics[0], "k8s/daemon_set/current_scheduled_nodes",
	                           ^
collection/daemonsets_test.go:50:29: copylocks: call of testutils.AssertMetrics copies lock value: github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1.Metric contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
	testutils.AssertMetrics(t, *rm.metrics[1], "k8s/daemon_set/desired_scheduled_nodes",
	                           ^
collection/deployments_test.go:38:30: copylocks: call of testutils.AssertResource copies lock value: github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1.Resource contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
	testutils.AssertResource(t, *rm.resource, k8sType,
	                            ^
collection/resourcequotas_test.go:47:39: copylocks: call of testutils.AssertMetricsWithLabels copies lock value: github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1.Metric contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
	testutils.AssertMetricsWithLabels(t, *actualResourceMetrics[0].metrics[0], "k8s/resource_quota/hard_limt",
	                                     ^
collection/resourcequotas_test.go:50:39: copylocks: call of testutils.AssertMetricsWithLabels copies lock value: github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1.Metric contains google.golang.org/protobuf/internal/impl.MessageState contains sync.Mutex (govet)
	testutils.AssertMetricsWithLabels(t, *actualResourceMetrics[0].metrics[1], "k8s/resource_quota/used",

**Comment by bogdandrutu** [comment]
@nilebox almost there, few proto deprecated usages left :)

**Comment by nilebox** [comment]
@bogdandrutu I've switched to `gogo/protobuf` to fix the lint errors. If that causes any issues, we could just add an ignore lint comment instead.

**Comment by bogdandrutu** [comment]
@nielbox thanks for doing all of this

**Comment by nilebox** [comment]
@bogdandrutu No worries.
CircleCI seems tired (load tests are failing), I've asked @james-bebbington to re-run it a few times until they succeed :)

The code coverage failure can be ignored, I suppose.

**Comment by bogdandrutu** [comment]
2020/07/24 02:16:09 Starting load generator at 10000 items/sec.
2020/07/24 02:16:12 Agent RAM (RES):  35 MiB, CPU: 0.0% | Sent:   207,200 items | Received:   206,500 items (68,838/sec)
2020/07/24 02:16:15 Agent RAM (RES):  59 MiB, CPU:29.3% | Sent:   417,200 items | Received:   416,500 items (69,418/sec)
2020/07/24 02:16:15 Performance error: RAM consumption is {61} MiB, max expected is 60 MiB
2020/07/24 02:16:15 Gracefully terminating Agent pid=18557, sending SIGTEM...

I think you need a bit more memory for OC metrics test, maybe increase that to 70?

**Review by bogdandrutu — APPROVED** [review]

**Pull request merged**